content/go/CDN_services.go.md

```diff
@@ -0,0 +1,325 @@
+---
+title: "Using IBM Cloud CDN Services"
+description: "CDN_services.go"
+date: "2020-10-07"
+classes: 
+    - "SoftLayer_Network_CdnMarketplace_Configuration_Mapping"
+    - "SoftLayer_Network_CdnMarketplace_Configuration_Mapping_Path"
+    - "SoftLayer_Container_Network_CdnMarketplace_Configuration_Input"
+tags:
+    - "cdn"
+---
+
+
+### Base Setup
+
+All the examples below will require this small bit of code, which imports a few libraries and sets up the SoftLayer client.
+
+
+```go
+package main
+import (
+    "encoding/json"
+    "fmt"
+    "github.com/softlayer/softlayer-go/sl"
+    "github.com/softlayer/softlayer-go/services"
+    "github.com/softlayer/softlayer-go/session"
+)
+
+// You can pass username, password into New() here, otherwise they are read from the ~/.softlayer configuration file
+sess := session.New()
+```
+
+### [ListVendors](/reference/services/SoftLayer_Network_CdnMarketplace_Vendor/listVendors/)
+This will list out the vendors available for CDN orders.
+
+```go
+func ListVendors() {
+    service := services.GetNetworkCdnMarketplaceVendorService(sess)
+
+    result, err := service.ListVendors()
+    
+    if err != nil {
+        fmt.Printf("\nAPI Error:\n - %s\n", err)
+        return
+    }
+
+    // Following helps to print the result in json format.
+    jsonFormat1, jsonErr1 := json.MarshalIndent(result, "", "     ")
+    if jsonErr1 != nil {
+        fmt.Println(jsonErr1)
+        return
+    }
+    fmt.Println(string(jsonFormat1))
+}
+```
+
+### [VerifyCdnAccountExists](/reference/services/SoftLayer_Network_CdnMarketplace_Account/verifyCdnAccountExists/)
+
+```go
+func VerifyCdnAccountExists() {
+    service := services.GetNetworkCdnMarketplaceAccountService(sess)
+    
+    // vendor name would be from the ListVendors()
+    vendor := sl.String("akamai")
+
+    result, err := service.VerifyCdnAccountExists(vendor)
+    if err != nil {
+        fmt.Printf("\nAPI Error:\n - %s\n", err)
+        return
+    }
+
+    // Following helps to print the result in json format.
+    jsonFormat1, jsonErr1 := json.MarshalIndent(result, "", "     ")
+    if jsonErr1 != nil {
+        fmt.Println(jsonErr1)
+        return
+    }
+    fmt.Println(string(jsonFormat1))
+}
+```
+
+### [CreateDomainMapping](/reference/services/SoftLayer_Network_CdnMarketplace_Configuration_Mapping/createDomainMapping/)
```

Comment:
Does this function actually order CDN? Or just create another mapping? I think we need an example on how to place an order as well, at least according to https://cloud.ibm.com/docs/CDN?topic=CDN-code-examples-using-the-cdn-api#example-code-to-place-order

Reply:
I also made a few changes to the structure of this article, so make sure to pull them in locally.

Reply:
> Does this function actually order CDN? Or just create another mapping? I think we need an example on how to place an order as well, at least according to https://cloud.ibm.com/docs/CDN?topic=CDN-code-examples-using-the-cdn-api#example-code-to-place-order
Yeah, we does need an example on how to place an order, this API is needed for the customer firstly order CDN and his CDN account has not been created.